test/integration/shell_int_test.go

```diff
@@ -276,6 +278,53 @@ func (suite *ShellIntegrationTestSuite) SetupRCFile(ts *e2e.Session) {
 	suite.Require().NoError(err)
 }
 
+func (suite *ShellIntegrationTestSuite) TestNestedShellNotification() {
+	if runtime.GOOS == "windows" {
+		return // cmd.exe does not have an RC file to check for nested shells in
+	}
+	suite.OnlyRunForTags(tagsuite.Shell)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cfg, err := config.New()
+	suite.Require().NoError(err)
+
+	ss := subshell.New(cfg)
+	err = subshell.ConfigureAvailableShells(ss, cfg, nil, sscommon.InstallID, true) // mimic installer
+	suite.Require().NoError(err)
+
+	rcFile, err := ss.RcFile()
+	suite.Require().NoError(err)
+	suite.Require().FileExists(rcFile)
+	suite.Require().Contains(string(fileutils.ReadFileUnsafe(rcFile)), "State Tool is operating on project")
+
+	cp := ts.Spawn("checkout", "ActiveState-CLI/small-python")
+	cp.Expect("Checked out project")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnWithOpts(
+		e2e.WithArgs("shell", "small-python"),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+	)
+	cp.Expect("Activated")
+	suite.Assert().NotContains(cp.TrimmedSnapshot(), "State Tool is operating on project")
+
+	binary := ss.Binary() // platform-specific shell (zsh on macOS, bash on Linux, etc.)
+	// Cannot run bare binary because it will not load the rcFile in ts.Dirs.HomeDir.
+	// Instead, tell the shell to load rcFile. This is not needed in a non-test environment.
+	switch ss.Shell() {
+	case bash.Name:
+		binary = fmt.Sprintf("%s --rcfile %s", binary, rcFile)
+	case zsh.Name:
+		binary = fmt.Sprintf("ZDOTDIR=%s %s", filepath.Dir(rcFile), binary)
```

Comment:
We shouldn't be specifying an RC file here. The use-case we're testing is the user running a new subshell (eg. just running `bash`), so this test should do that.

Reply:
There's a comment 6 lines above explaining why I need it for the test: https://github.com/ActiveState/cli/pull/2710/files/3cfa3a53d6ba9ccb67ab832f4208c24e9df8c332#diff-5a8d1281306bbab65982eb86bcf4706c963f88bb9821aeb35bebacd03d2ea8b6R313-R314

Running a bare `zsh` sources the machine's *~/.zshrc*, which is unmodified because our tests use a test-specific user home directory.

Reply:
That seems like a limitation in our test suite though. I just tried `HOME=/foo zsh` and it tried to source its rc file from `/foo`. I'd say we should do that for our test suites, but considering that may be too big a change for this PR I suggest just doing it for these tests.